src/ferc1_eia_match/candidate_set_creation.py

```diff
@@ -0,0 +1,344 @@
+"""Blocking methods to create a candidate set of tuples matches."""
+import logging
+
+import faiss
+import numpy as np
+import pandas as pd
+import sqlalchemy as sa
+from scipy.sparse import hstack, issparse
+from sklearn.feature_extraction.text import TfidfVectorizer
+from sklearn.preprocessing import MinMaxScaler
+
+logger = logging.getLogger(__name__)
+
+
+class DataframeEmbedder:
```

Comment:
This class takes the left and right dataframe as input, and creates an embedding matrix of the dataframes that are ready to be inputted into the similarity search.

The string columns (plant_name, utility_name, fuel_type_code_pudl) are vectorized with TF-IDF and null values are filled with the empty string. For fuel_type_code_pudl, a little bit of imputation is attempted, where if there is the FTCP in the plant_name, then the null value is filled in. E.g. "Miller Hydro"

The numeric columns (installation_year, construction_year, capacity_mw) are normalized between 0 and 1 and null values for filled in with the median value. If an installation_year is null but construction_year is not, then the installation_year is filled in with the construction_year value, and vice versa.

After vectorizing all the columns, the embeddings are stored in `col_embedding_dict` and then concatenated all together into one matrix using equal weight for each column.